src/Client.js

```diff
@@ -0,0 +1,110 @@
+import Api from './core/Api';
+import SocketHandler from './core/SocketHandler';
+import { Event } from './core/ChainEvent';
+import { find } from 'lodash';
+
+export default class Client {
+  constructor(
+    apiUrl = 'http://localhost:4000',
+    socketUrl = 'ws://127.1:4000/socket'
+  ) {
+    this._api = new Api(apiUrl);
+    this._socket = new SocketHandler(socketUrl);
+  }
+
+  async init() {
+    await this._socket.init();
+    await this.once('api', Event.API_JOIN);
+  }
+
+  api() {
+    return this._api;
+  }
+
+  socket() {
+    return this._socket;
+  }
+
+  channel(id) {
+    return this._socket.channel(id);
+  }
+
+  channels() {
+    return this._socket._channels;
```

Comment:
it's a code smell when you're accessing the underscore-prefixed properties of another object (in this case `_channels`), because underscore-prefixed properties are named like that to indicate that they should be treated as private. so you either want to remove the underscore, to indicate that it's for use outside the object, or create a method that provides limited access to it. this goes for `delete this._socket._channels[id]` below as well.